storage/Devicegraph.cc

```diff
@@ -43,13 +44,15 @@ namespace storage
 
 
     WrongNumberOfParents::WrongNumberOfParents(size_t seen, size_t expected)
-	: Exception(sformat("wrong number of parents, seen '%zu', expected '%zu'", seen, expected))
+	: Exception(sformat("wrong number of parents, seen '%zu', expected '%zu'", seen, expected),
+		    Silencer::is_any_active() ? DEBUG : WARNING)
```

Comment:
Currently it makes some exception silent, e.g. DeviceHasWrongType but not NullPtrException.
And some exception log ERROR and some WARNING. We will have to see how this proves in the future, so currently no contract.